File: app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/items/detail/ItemSummaryFragment.java
package com.gatheringhallstudios.mhworlddatabase.features.items.detail;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.gatheringhallstudios.mhworlddatabase.R;
import com.gatheringhallstudios.mhworlddatabase.data.models.Item;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ItemSummaryFragment extends Fragment {

    //@BindView(R.id.item_icon)
    //ImageView itemIcon;

    @BindView(R.id.item_name)
    TextView itemName;

    @BindView(R.id.item_description)
    TextView itemDescription;

    @BindView(R.id.item_data_layout)
    LinearLayout itemDataLayout;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup parent,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_item_summary, parent, false);
        ButterKnife.bind(this, view);

        ItemDetailViewModel viewmodel = ViewModelProviders.of(getParentFragment()).get(ItemDetailViewModel.class);
        viewmodel.getItem().observe(this, this::populateItem);

        return view;
    }

    private void populateItem(Item item) {

        //Set the summary information
        //itemIcon.setIcon(setsomeicon);
        itemName.setText(item.getName());
        itemDescription.setText(item.getDescription());

        // Find target views manually since we can't run ButterKnife.bind more than once.
        TextView buyPriceValue = itemDataLayout.findViewById(R.id.buy_price_value);
        TextView sellPriceValue = itemDataLayout.findViewById(R.id.sell_price_value);
        TextView carryCapacityValue = itemDataLayout.findViewById(R.id.carry_capacity_value);
        TextView rarityValue = itemDataLayout.findViewById(R.id.rarity_value);

        bindItemDataValue(buyPriceValue, item.getData().getBuy_price());
        bindItemDataValue(sellPriceValue, item.getData().getSell_price());
        bindItemDataValue(carryCapacityValue, item.getData().getCarry_limit());
        bindItemDataValue(rarityValue, item.getData().getRarity());
    }

    /**
     * Binds item data values from the itemView object to the view
     */
    private void bindItemDataValue(TextView textView, int value) {
        //Replace null/0 value with -
        if(value == 0) {
            textView.setText("-");
        } else {
            textView.setText(Integer.toString(value));

        }
    }
}

File: app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/items/detail/ItemSummaryFragment.kt
package com.gatheringhallstudios.mhworlddatabase.features.items.detail

import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.gatheringhallstudios.mhworlddatabase.R
import com.gatheringhallstudios.mhworlddatabase.assets.AssetLoader
import com.gatheringhallstudios.mhworlddatabase.data.models.Item
import com.gatheringhallstudios.mhworlddatabase.setActivityTitle
import com.gatheringhallstudios.mhworlddatabase.util.getDrawableCompat
import com.gatheringhallstudios.mhworlddatabase.databinding.FragmentItemSummaryBinding


/**
 * Binds item data values from the itemView object to the view
 */
private fun evaluateValue(value: Int?) = when(value) {
    0, null -> "-"
    else -> value.toString()
}

class ItemSummaryFragment : androidx.fragment.app.Fragment() {
    private var _binding: FragmentItemSummaryBinding? = null
    private val binding get() = _binding!!


    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        _binding = FragmentItemSummaryBinding.inflate(inflater, parent, false)

        val viewmodel = ViewModelProvider(parentFragment!!).get(ItemDetailViewModel::class.java)
        viewmodel.item.observe(this, Observer(::populateItem))

        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    private fun populateItem(item: Item?) {
        if (item == null) return

        //Set the summary information
        binding.itemHeader.setIconDrawable(AssetLoader.loadIconFor(item))
        binding.itemHeader.setTitleText(item.name)
        binding.itemHeader.setDescriptionText(item.description)

        binding.rarityValue.setTextColor(AssetLoader.loadRarityColor(item.rarity))
        binding.rarityValue.text = getString(R.string.format_rarity, item.rarity)

        binding.buyPriceValue.text = evaluateValue(item.buy_price)
        binding.carryCapacityValue.text = evaluateValue(item.carry_limit)

        // Set sell value. Swaps to research points if research points are available
        if (item.sell_price == 0 && item.points > 0) {
            binding.sellPriceValue.text = item.points.toString()
            binding.sellPriceIcon.setImageDrawable(context?.getDrawableCompat(R.drawable.ic_ui_research_points))
        } else {
            binding.sellPriceValue.text = evaluateValue(item.sell_price)
        }

    }
}
